core_types/src/entry/cap_entries.rs

```diff
@@ -62,18 +62,26 @@ pub type CapTokenValue = Address;
 /// a collection functions by zome name that are authorized within a capability
 pub type CapFunctions = BTreeMap<String, Vec<String>>;
 
-/// System entry to hold a capability token for use as a caller
+/// System entry to hold a capability token claim for use as a caller
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, DefaultJson)]
-pub struct CapToken {
+pub struct CapTokenClaim {
+    id: String,
+    grantor: Address,
     token: CapTokenValue,
 }
 
-impl CapToken {
-    pub fn new(token: CapTokenValue) -> Self {
-        CapToken { token }
+impl CapTokenClaim {
+    pub fn new(id: String, grantor: Address, token: CapTokenValue) -> Self {
+        CapTokenClaim { id, grantor, token }
+    }
+    pub fn token(&self) -> CapTokenValue {
+        self.token.clone()
+    }
+    pub fn id(&self) -> String {
+        self.id.clone()
     }
-    pub fn token(self) -> CapTokenValue {
-        self.token
+    pub fn grantor(&self) -> Address {
```

Comment:
What happen to CapTokenValue?

Reply:
good point, I think I should delete it completely because it's allways an address and I don't think the type system can help us in any way around when we are treating an address as a token and when not.